php/src/class-plugin-settings.php

```diff
@@ -0,0 +1,569 @@
+<?php
+/**
+ * Generate the plugin settings UI.
+ *
+ * @package XWP\Site_Performance_Tracker
+ * @license http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace XWP\Site_Performance_Tracker;
+
+/**
+ * Class Plugin_Settings
+ */
+class Plugin_Settings {
+
+	/**
+	 * PerformanceObserver default chance of sending performance metrics to analytics.
+	 *
+	 * Set to 100% which sends analytics on every request.
+	 *
+	 * @var int
+	 */
+	const TRACKING_DEFAULT_CHANCE = 1;
+
+	/**
+	 * WordPress admin settings page ID.
+	 *
+	 * @var string
+	 */
+	const SETTINGS_PAGE_ID = 'site_performance_tracker';
+
+	/**
+	 * Option key for web vitals settings store.
+	 *
+	 * @var string
+	 */
+	const SETTINGS_VITALS_SECTION_ID = 'spt_settings';
+
+	/**
+	 * Key used for theme supports config.
+	 *
+	 * @var string
+	 */
+	const THEME_SUPPORTS_VITALS_KEY = 'site_performance_tracker_vitals';
+
+	const SETTINGS_VITALS_FIELD_GTAG_ID = 'gtag_id';
+
+	const SETTINGS_VITALS_FIELD_DIMENSION_MEASUREMENT_VERSION = 'measurementVersion';
+
+	const SETTINGS_VITALS_FIELD_DIMENSION_EVENT_META = 'eventMeta';
+
+	const SETTINGS_VITALS_FIELD_DIMENSION_EVENT_DEBUG = 'eventDebug';
+
+	const SETTINGS_VITALS_FIELD_TRACKING_RATIO = 'web_vitals_tracking_ratio';
+
+	/**
+	 * Option page ID.
+	 *
+	 * @var string
+	 */
+	protected $option_page_id;
+
+	/**
+	 * Theme Support setting fetcher.
+	 *
+	 * @var Theme_Support_Setting
+	 */
+	protected $theme_settings;
+
+	/**
+	 * Settings page.
+	 *
+	 * @param Plugin $plugin Plugin instance.
+	 */
+	public function __construct( $plugin ) {
+		$this->plugin = $plugin;
+
```

Comment:
Apart from the `$this->plugin` property being declared dynamically, it's also not being used at all, so I assume this is just a leftover.
```suggestion
	 */
	public function __construct() {
```